src/theory/theory_model_builder.cpp

```diff
@@ -47,6 +47,14 @@ bool TheoryEngineModelBuilder::isAssignable(TNode n)
       return !n.getType().isFunction();
     }
   }
+  else if (n.getKind() == kind::FLOATINGPOINT_COMPONENT_SIGN
+           || n.getKind() == kind::FLOATINGPOINT_COMPONENT_SIGNIFICAND)
```

Comment:
I don't think significand should be treated the same.  The auxiliary constraints that keep the components in sync should always mention it so the BV solver should always return a term for it.  Also it is not correct to simply pick any value.